Added header control topics (#209)
agentgateway/website: assets/agw-docs/pages/traffic-management/header-control/early-request-header-modifier.md

Early request header modification allows you to add, set, or remove HTTP request headers at the listener level, before route selection and other request processing occurs.

This capability is especially useful for security and sanitization use cases, where you want to ensure that sensitive headers cannot be faked by downstream clients and are only set by trusted components such as external authentication services.

> Early request header modification is configured on an `{{< reuse "agw-docs/snippets/trafficpolicy.md" >}}` using the `transformation` field. This policy is attached directly to a proxy and applies header mutations before route selection.

Reviewer: ```suggestion
Early request header modification is configured on an `{{< reuse "agw-docs/snippets/trafficpolicy.md" >}}` using the `transformation` field. This policy is attached directly to a proxy and applies header mutations before route selection. You can choose between the following header operations: 
```

> The configuration uses the AgentgatewayPolicy and supports the following operations:

Author: ```suggestion
The configuration uses the {{< reuse "agw-docs/snippets/trafficpolicy.md" >}} and supports the following operations:
```

   ```json {linenos=table,hl_lines=[12,13,14],linenostart=1}
   {
     "headers": {
       "Accept": [
         "*/*"
       ],
       "Host": [
         "transformation.example"
       ],
       "User-Agent": [
         "curl/8.7.1"
       ],
       "X-User-Id": [
         "reserved-user"
       ]
     }
   }
   ```

3. Create an AgentgatewayPolicy with a transformation to remove the `x-user-id` header. You can choose to apply the removal on the HTTPRoute or the Gateway with an AgentgatewayPolicy. 

   {{< tabs tabTotal="2" items="Route-level,Gateway-level" >}}
   {{% tab tabName="HTTPRoute (EnterpriseAgentgatewayPolicy)" %}}

   ```yaml
   kubectl apply -f- <<EOF
   apiVersion: {{< reuse "agw-docs/snippets/trafficpolicy-apiversion.md" >}}
   kind: {{< reuse "agw-docs/snippets/trafficpolicy.md" >}}
   metadata:
     name: remove-reserved-header
     namespace: httpbin
   spec:
     targetRefs:
       - group: gateway.networking.k8s.io
         kind: HTTPRoute
         name: httpbin-route
     traffic:
       transformation:
         request:
           remove:
             - x-user-id
   EOF
   ```

   {{% /tab %}}

>    {{% tab tabName="Gateway listener" %}}

Reviewer: ```suggestion
   {{% tab tabName="Gateway-level" %}}
```

           remove:
             - x-user-id
   EOF
   ```
   {{% /tab %}}
   {{< /tabs >}}

4. Repeat the test request to the sample httpbin app. The `x-user-id` header is no longer present in the response.

   {{< tabs items="Cloud Provider LoadBalancer,Port-forward for local testing" tabTotal="2" >}}
   {{% tab tabName="Cloud Provider LoadBalancer" %}}  
   ```sh
   curl -i http://$INGRESS_GW_ADDRESS:8080/headers -H "host: transformation.example" -H "x-user-id: reserved-user"
   ```
   {{% /tab %}}
   {{% tab tabName="Port-forward for local testing"%}}
   ```sh
   curl -i localhost:8080/headers -H "host: transformation.example" -H "x-user-id: reserved-user"
   ```
   {{% /tab %}}
   {{< /tabs >}}
   Example output: Note that the `X-User-Id` header is present in the request.

   ```json {linenos=table,hl_lines=[12,13],linenostart=1}
   {
     "headers": {
       "Accept": [
         "*/*"
       ],
       "Host": [
         "transformation.example"
       ],
       "User-Agent": [
         "curl/8.7.1"
       ]
     }
   }
   ```

## Cleanup

{{< reuse "agw-docs/snippets/cleanup.md" >}} Run the following commands.

```sh
kubectl delete httproute httpbin-route -n httpbin

> kubectl delete AgentgatewayPolicy remove-reserved-header -n {{< reuse "agw-docs/snippets/namespace.md" >}}

Reviewer: ```suggestion
kubectl delete {{< reuse "agw-docs/snippets/trafficpolicy.md" >}} remove-reserved-header -n {{< reuse "agw-docs/snippets/namespace.md" >}}
```

> kubectl delete AgentgatewayPolicy remove-reserved-header -n httpbin

Reviewer: ```suggestion
kubectl delete {{< reuse "agw-docs/snippets/trafficpolicy.md" >}} remove-reserved-header -n httpbin
```
